docs: Document Typography block category in the GrapesJS Email Builder (#838)
mautic/user-documentation: docs/builders/email_landing_page.rst

Reviewer: Give align center for the image.

```suggestion
.. image:: images/editorfonts.jpg
   :width: 280
   :alt: The Fonts in Style Manager of Typography
```

#. Content blocks. You can populate your newsletter with these content blocks. Each block has specific layout, settings and design.

.. vale off

Typography blocks
*****************

When you edit an MJML Email in the GrapesJS Builder, the blocks sidebar includes a dedicated Typography category. It groups ready-made text blocks so you can add headings and body text without styling each element by hand. The Typography category sits between the Sections and Blocks categories in the sidebar. It's collapsed by default, so select it to expand the blocks.

.. image:: images/editor_typography_blocks.png
  :width: 280

>   :alt: Screenshot of the Typography block category in the GrapesJS Builder blocks sidebar

Reviewer: Fix the indentation.

```suggestion
.. image:: images/editor_typography_blocks.png
   :width: 280
   :alt: The Typography category in the GrapesJS Builder blocks sidebar
```

The Typography category contains these blocks:

* Text - a plain text block for body copy
* H1, H2, H3, H4 - heading blocks in decreasing size

> * Subtitle - a smaller, italic block for supporting text

Reviewer: ```suggestion
* **Text** - a plain text block for body copy
* **H1**, **H2**, **H3**, **H4** - heading levels
* **Subtitle** - a smaller, italic block for supporting text
```

> The Text block now lives in the Typography category instead of its previous location, but it works exactly as before.

Reviewer: I'm not sure if we this wording is great. It might be helpful for existing users, but could be confusing for new users as it's mentioning the previous version.

Maybe something like "Since version 7.2, the Text block lives in the Typography category, but it works the same."

When you drag an H1, H2, H3, H4, or Subtitle block onto the canvas, Mautic opens the inline text editor and selects the placeholder text right away, so you can type over it immediately.

> Each heading and Subtitle block comes with default sizing and styling, such as a bold 32px H1 or an italic Subtitle. If your Theme defines its own typography tokens, the blocks use the Theme's styles instead of these defaults, so the same block can look different across Themes.

Reviewer: ```suggestion
Each heading and Subtitle block comes with default sizing and styling, such as a bold 32px for H1 or an italic subtitle. If your Theme defines its own typography tokens, the blocks use the Theme's styles instead of these defaults, so the same block can look different across Themes.
```

What does "so the same block can look different across Themes." mean?

#. Content blocks. You can populate your newsletter with these content blocks. Each block has specific layout, settings and design.

> .. vale off

Reviewer: Remove the vale off/on statements.

Typography blocks
*****************

> When you edit an MJML Email in the GrapesJS Builder, the blocks sidebar includes a dedicated Typography category. It groups ready-made text blocks so you can add headings and body text without styling each element by hand. The Typography category sits between the Sections and Blocks categories in the sidebar. It's collapsed by default, so select it to expand the blocks.

Reviewer: - Move the "since version 7.2" here.
- Inject the note admonition at the end in this paragraph.
- Remove sentence after the comma in the 1st sentence because it's a repetition. We already have it here: The Typography category sits between the Sections and Blocks categories in the sidebar. 

```suggestion
.. note::

   Since version 7.2, the Text block lives in the Typography category, but it works the same.

The Typography category is **only available** when you edit MJML-based Emails in the GrapesJS Builder. It groups ready-made text blocks so you can add headings and body text without styling each element by hand. The Typography category sits between the Sections and Blocks categories in the sidebar. It collapses by default, so click it to expand the blocks.
```

.. image:: images/editor_typography_blocks.png
   :width: 280
   :alt: The Typography category in the GrapesJS Builder blocks sidebar

The Typography category contains these blocks:

* **Text** - a plain text block for body copy
* **H1**, **H2**, **H3**, **H4** - heading levels
* **Subtitle** - a smaller, italic block for supporting text

> Since version 7.2, the Text block lives in the Typography category, but it works the same.

Reviewer: I'm in favor of moving this to the top for direct info. I feel like this is an orphan between the paragraphs. It' cuts the flow.

When you drag an H1, H2, H3, H4, or Subtitle block onto the canvas, Mautic opens the inline text editor and selects the placeholder text right away, so you can type over it immediately.

Each heading and Subtitle block comes with default sizing and styling, such as a bold 32px for H1 or an italic Subtitle. If your Theme defines its own typography tokens, the blocks use the Theme's styles instead of these defaults. Since each Theme can define its own tokens, the same block can render with different sizes, weights, or colors depending on which Theme you apply.

.. note::

>    The Typography category is only available when you edit MJML-based Emails in the GrapesJS Builder.

Reviewer: Remove this as we move it to the 1st paragraph.

#. Content blocks. You can populate your newsletter with these content blocks. Each block has specific layout, settings and design.

Typography blocks
*****************

> When you edit an MJML Email in the GrapesJS Builder, the blocks sidebar includes a dedicated Typography category. It groups ready-made text blocks so you can add headings and body text without styling each element by hand. The Typography category sits between the Sections and Blocks categories in the sidebar. It's collapsed by default, so select it to expand the blocks.

Author: Documented the new Typography block category and its H1–H4/Subtitle blocks from this PR. The category label, block labels, and ordering come from `Translations/en_US/javascript.ini` (`grapesjsbuilder.categoryTypographyLabel`, `grapesjsbuilder.blocks.typography.*`) and the `order: -0.5`/`open: false` registration in `grapesjs-mjmlThemeTokens/blocks.js`, which place Typography between Sections and Blocks, collapsed by default.

Source: https://github.com/mautic/mautic/pull/15694

.. image:: images/editor_typography_blocks.png
  :width: 280
  :alt: Screenshot of the Typography block category in the GrapesJS Builder blocks sidebar

The Typography category contains these blocks:

* Text - a plain text block for body copy
* H1, H2, H3, H4 - heading blocks in decreasing size
* Subtitle - a smaller, italic block for supporting text

The Text block now lives in the Typography category instead of its previous location, but it works exactly as before.

> When you drag an H1, H2, H3, H4, or Subtitle block onto the canvas, Mautic opens the inline text editor and selects the placeholder text right away, so you can type over it immediately.

Author: The instant-edit-on-drop behavior for H1–H4 and Subtitle comes from `grapesjs-mjmlThemeTokens/blocks.js`, where each heading/subtitle block sets `activate: true` and a `block:drag:stop` listener selects the placeholder text for the `typographyBlockIds` set (`mj-heading-1..4`, `mj-subtitle`). The relocated Text block uses `action: 'patch'` without `activate`, so I scoped the auto-select statement to the heading and Subtitle blocks only.

Source: https://github.com/mautic/mautic/pull/15694

#. Content blocks. You can populate your newsletter with these content blocks. Each block has specific layout, settings and design.

Typography blocks

> *****************

Reviewer: ```suggestion
Typography
**********
```

#. Content blocks. You can populate your newsletter with these content blocks. Each block has specific layout, settings and design.

Typography blocks

> *****************
> 
> .. note::
> 
>    Since Mautic 7.2, the GrapesJS Builder includes the Typography category.

Reviewer: ```suggestion
.. note::

   .. vale off
   
   Since Mautic 7.2, the GrapesJS Builder includes the Typography category.
   
   .. vale on
```

.. vale off

The Typography category is **only available** when you edit MJML-based Emails in the GrapesJS Builder. It groups ready-made text blocks so you can add headings and body text without styling each element by hand. You can find it between the Sections and Blocks categories in the sidebar. It starts closed by default, so click it to expand the category.

.. image:: images/editor_typography_blocks.png
   :width: 280

>    :alt: The Typography category in the GrapesJS Builder blocks sidebar

Reviewer: Give align center for the image.

#. Content blocks. You can populate your newsletter with these content blocks. Each block has specific layout, settings and design.

Typography
**********

.. note::

   .. vale off

   Since Mautic 7.2, the GrapesJS Builder includes the Typography category.

   .. vale on

.. vale off

The Typography category is **only available** when you edit MJML-based Emails in the GrapesJS Builder. It groups ready-made text blocks so you can add headings and body text without styling each element by hand. You can find it between the Sections and Blocks categories in the sidebar. It starts closed by default, so click it to expand the category.

> .. image:: images/editor_typography_blocks.png
>    :width: 280
>    :align: center
>    :alt: The Typography category in the GrapesJS Builder blocks sidebar

Reviewer: Add some space after the image.

```suggestion
.. image:: images/editor_typography_blocks.png
   :width: 280
   :align: center
   :alt: The Typography category in the GrapesJS Builder blocks sidebar
   
|
```

#. Content blocks. You can populate your newsletter with these content blocks. Each block has specific layout, settings and design.

.. _typography blocks section:

Typography blocks
*****************

.. note::

>    Since version 7.2, the Text block lives in the Typography category, but it works the same.

Reviewer: Please investigate the codebase across branches. When did we first introduce typography category? If we've introduced it in earlier version than 7, remove this note admonition.

The Typography category is **only available** when you edit MJML-based Emails in the GrapesJS Builder. It groups ready-made text blocks so you can add headings and body text without styling each element by hand. The Typography category sits between the Sections and Blocks categories in the sidebar. It collapses by default, so click it to expand the blocks.

.. image:: images/editor_typography_blocks.png
   :width: 280
   :alt: The Typography category in the GrapesJS Builder blocks sidebar

> The Typography category contains these blocks:

Reviewer: ```suggestion
.. vale off

The Typography category is **only available** when you edit MJML-based Emails in the GrapesJS Builder. It groups ready-made text blocks so you can add headings and body text without styling each element by hand. You can find it between the Sections and Blocks categories in the sidebar. It starts closed by default, so click it to expand the category.

.. image:: images/editor_typography_blocks.png
   :width: 280
   :alt: The Typography category in the GrapesJS Builder blocks sidebar

The Typography category contains these blocks:

.. vale on
```

Use the Component typography controls to fine-tune headings, paragraphs, and other text elements in legacy Themes that lack modern styling flexibility.

Custom fonts
------------

You can extend the **Typography** > **Fonts** list to include custom fonts.

.. image:: images/editorfonts.jpg
   :width: 280
   :align: center

>    :alt: The Fonts in Style Manager of Typography

Reviewer: Add a space after image here, too.

To learn more about creating Themes, see :doc:`/builders/creating_themes`.

.. _typography styling section:

> Typography styling

Reviewer: - Move this section as part of "Typography blocks".
- Make the heading an H3.
- Adjust the wording for the opening and remove the reference as it's becoming part of the "Typography blocks".
